rust/xaynet-analytics/src/controller.rs

```diff
@@ -121,18 +146,39 @@ impl AnalyticsController {
         )
     }
 
-    // TODO: review and debug this method during https://xainag.atlassian.net/browse/XN-1560
+    /// This method implements a sliding 'time window' of self.send_data_frequency duration, to check whether we have
+    /// already sent data in the current window, or not.
+    ///
+    /// An alternative implementation could be based on simply checking whether:
+    /// last_time_data_sent > Utc::now() - self.send_data_frequency
+    ///
+    /// In the current implementation, it might be easier to then group the aggregated data on the coordinator side,
+    /// to then be displayed in the UI, especially if self.send_data_frequency = Duration::hours(24).
+    ///
+    /// The more dynamic approach however implies that if, for example, self.send_data_frequency = Duration::hours(6),
+    /// and the last time we sent the data was at 5AM, we would be able to send again at 7AM, while with the simpler solution
+    /// we wouldn't be able to send again until 11AM.
+    ///
+    /// The correct approach to be chosen should very much depend on the amount of data available for aggregation,
+    /// and it's possible that Self::MAX_SEND_DATA_FREQUENCY_HOURS should be increased to more than 24.
+    /// Only once it's more clear how the aggregation will work on the coordinator side, there will be more information
+    /// to decide the approach here.
     fn did_send_already_in_this_period(&self) -> bool {
```

Comment:
it might also be worth mentioning that in the current implementation, ideally `send_data_frequency` divides `MAX_SEND_DATA_FREQUENCY_HOURS` exactly. otherwise, the last window will be smaller than the ones before it (which may be fine, but I think is something noteworthy).